demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -195,8 +204,12 @@ int main(int argc, char *argv[]) {
         if (static_cast<int>(labels.size()) != num_classes) {
             if (static_cast<int>(labels.size()) == (num_classes - 1))  // if network assumes default "background" class, having no label
                 labels.insert(labels.begin(), "fake");
-            else
+            else {
+                slog::info << "The number of labels (" << labels.size() << ") "
+                    << "is different from numbers of model classes (" << num_classes << "). "
+                    << "Labels are omitted." << slog::endl;
```

Comment:
I have an idea it is better to throw an exception and fail instead of trying to pseudo fix labels. A user can do the same by removing `-labels` and this is more explicit.

Reply:
Fair enough, I just tried to save old behavior.

Reply:
I'm not sure how to choose an exception type, but given that the demo uses `logic_error` mostly, you should throw `logic_error`.

Reply:
Done. There are both appropriate, but in ssd_demo all errors are logic, so be it. I dont change it in yolo_demo, since there are presented both type of errors.